### PR TITLE
Adjust parameters for callbacks in some libraries, and some other misc. changes.

### PR DESCRIPTION
**Pull request by bythos14**
Added missing param to ScePowerCallback.

Adjusted params for callbacks and fixed some indentations in SceLibJson.

Fixed a couple typos from 8ebd0b29 in fcntl.h.

Renamed `common` argument of SceKernelCallbackFunction to `userData`. Defined unknown `arg` argument in sceKernelCreateCallback as `userData`.

### CODE REVIEW

**Comment by yne** [comment]
may I ask why `userData` seems more appropriate that just `data` ?
Because it point to a userland address ?

Just askin'

**Comment by bythos14** [comment]
There's not really much difference tbh. I defined it as 'User defined data' so `userData` just seemed like a good name. Honestly could have left it unchanged ¯\_(ツ)_/¯

File: include/psp2/json.h [hunk 1]
```diff
@@ -276,7 +276,7 @@ class Array
 	Array& operator=(const Array& arr);
 
 private:
-	void* impl_data; //!< Pointer to internal implementation data
+	void* impl_data; //!< Pointer to internal implementation data.
```
Comment (d3m3vilurr) [inline include/psp2/json.h:279]: userData and errorCode are `camelCase` but `impl_data` is `snake_case`.
could you choose one notation for this lib?

Comment (bythos14) [reply include/psp2/json.h]: Done.

**Pull request merged**